cg/store/api/find_business_data.py

```diff
@@ -550,3 +551,12 @@ def is_case_external(self, case_id: str) -> bool:
         """Returns True if all samples in a case have been sequenced externally."""
         case: Family = self.family(internal_id=case_id)
         return all(sample.application_version.application.is_external for sample in case.samples)
+
+    def get_case_by_internal_id(self, internal_id: str) -> Family:
+        """Get case by internal id."""
+
+        return apply_case_filter(
+            filter_functions=[CaseFilter.FILTER_BY_INTERNAL_ID],
+            cases=self._get_query(table=Family),
+            internal_id=internal_id,
+        ).first()
```

Comment:
This kind of code is starting to look very clean! Great work @seallard @Vince-janv @ChrOertlin @diitaz93